### PR TITLE
Support FantasyLand 1.x

### PR DESCRIPTION
**Pull request by Avaq**
* Prefixed method names
* Flipped `ap`

### CODE REVIEW

**Comment by Avaq** [comment]
I would like to ask the community whether there's anything I forgot in order to support FL-1.0?

**Comment by rpominov** [comment]
Prefixes and flipped `ap` is probably all there is in Fluture.

File: fluture.js [hunk 1]
```diff
@@ -701,15 +702,6 @@
...
 
-  //Creates a dispatcher for a unary method, but takes the object first rather than last.
-  function createInvertedUnaryDispatcher(method){
-    return function invertedUnaryDispatch(m, a){
-      if(arguments.length === 1) return unaryPartial(invertedUnaryDispatch, m);
-      if(m && typeof m[method] === 'function') return m[method](a);
-      error$invalidArgument(`Future.${method}`, 0, `have a "${method}" method`, m);
-    };
-  }
-
```
Comment (davidchambers) [inline fluture.js]: It's nice to see this go away. :)

File: test/fluture.test.js [hunk 2]
```diff
@@ -6,11 +6,11 @@ const Readable = require('stream').Readable;
...
-  bimap: 'bimap',
-  chain: 'chain',
-  ap: 'ap',
-  of: 'of'
+  map: 'fantasy-land/map',
+  bimap: 'fantasy-land/bimap',
+  chain: 'fantasy-land/chain',
+  ap: 'fantasy-land/ap',
+  of: 'fantasy-land/of'
 };
```
Comment (davidchambers) [inline test/fluture.test.js]: I suggest adding fantasy-land as a dev dependency and requiring it here to catch typos. :)

Comment (Avaq) [inline test/fluture.test.js]: Ahh, that's a great idea! That way we're ensuring compatibility with the naming conventions of a specific FL version, without having a hard dependency to it! :D

Comment (rpominov) [inline test/fluture.test.js]: Just curious why not actually add a hard dependency?

Comment (rpominov) [inline test/fluture.test.js]: I mean I can understand not wanting a peer dependency, but what bad about a normal dependency?

Comment (davidchambers) [inline test/fluture.test.js]: > Just curious why not actually add a hard dependency?

I don't see compelling reasons to do so: https://github.com/sanctuary-js/sanctuary-type-classes/pull/2#issuecomment-247303007.

Comment (Avaq) [inline test/fluture.test.js]: FantasyLand comes with its own dependency tree which seems to me like unnecessary overhead. It's not much of a big deal, but if there's no practical difference, or even much of a code-reuse advantage, I consider it cleaner not to depend on it.

Comment (rpominov) [inline test/fluture.test.js]: > FantasyLand comes with its own dependency tree which seems to me like unnecessary overhead.

Didn't know that, created a PR that fixes that https://github.com/fantasyland/fantasy-land/pull/168

Other points are valid though, I understand it better now, thanks!

File: README.md [hunk 3]
```diff
@@ -802,3 +807,4 @@ means butterfly in Romanian; A creature you might expect to see in Fantasy Land.
 [22]: https://github.com/Avaq/Fluture/wiki/Comparison-to-Promises
 [23]: https://vimeo.com/106008027
 [24]: https://github.com/fantasyland/fantasy-land#bifunctor
+[25]: https://github.com/fantasyland/fantasy-land/tree/1.0.0
```
Comment (davidchambers) [inline README.md]: Should this be `1.0.1`?

Comment (Avaq) [inline README.md]: I wish it could be `^1.0.0`, maybe I should just use `master` and pin it once a `2.0` comes out.

**Pull request merged**